examples/lighting-app/nrfconnect/main/include/Rpc.h

```diff
@@ -0,0 +1,66 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <kernel.h>
+
+#ifdef CONFIG_CHIP_PW_RPC
+#pragma message "Building with RPC support"
+#include "pw_sys_io_nrfconnect/init.h"
+
+namespace {
+#define RPC_STACK_SIZE 4096
+#define RPC_PRIORITY 5
+
+K_THREAD_STACK_DEFINE(rpc_stack_area, RPC_STACK_SIZE);
+struct k_thread rpc_thread_data;
+} // namespace
+
+namespace chip {
+namespace rpc {
+
+class LightingService;
+
+void RunRpcService(void *, void *, void *);
+
+inline k_tid_t Init()
```

Comment:
Why inline?

If not inline could you not remove the #ifdefs from this header altogether?

```
namespace chip {
namespace rpc {

extern k_tid_t Init();

}  // namespace rpc
}  // namespace chip
```

or am I missing something?

Simplifying the ifdefs at least would be nice. The namespace could be factored out (and just have the unnamed namespace inside namespace ::chip::rpc.

Reply:
The definition of Init() has to be somewhere since it is called from main(). Rpc.cpp is only added to the list of sources if CONFIG_CHIP_PW_RPC is defined. The definition could go there but then Rpc.cpp would have to be always compiled (which is OK) and have an #ifdef in it too. Basically we would be moving the #ifdef from the header to the .cpp. What's wrong with `inline`?

Reply:
There's nothing fundamentally wrong with inline but we should try not put the whole program in header files. It's cleaner to use headers to define the interface and define in the .cc file. These days we also have LTO which allows inlining without putting definitions in headers.

ifdef in .cc file is generally better; then we don't have to recompile other source files when the option changes (although the the Zephyr build system is not smart enough to take advantage of this).

Reply:
done